src/NextjsImage.ts

```diff
@@ -35,56 +26,37 @@ export interface NextjsImageProps extends NextjsBaseProps {
 /**
  * This lambda handles image optimization.
  */
-export class NextjsImage extends Function {
-  bucket: IBucket;
-
+export class NextjsImage extends NodejsFunction {
   constructor(scope: Construct, id: string, props: NextjsImageProps) {
-    const { lambdaOptions, bucket, isPlaceholder } = props;
-
-    const code = isPlaceholder
-      ? Code.fromInline(
-          "module.exports.handler = async () => { return { statusCode: 200, body: 'cdk-nextjs placeholder site' } }"
-        )
-      : Code.fromAsset(props.nextBuild.nextImageFnDir);
+    const { lambdaOptions, bucket } = props;
 
+    const nodejsFnProps = getCommonNodejsFunctionProps(scope);
     super(scope, id, {
-      // open-next image-optimization-function
-      // see: https://github.com/serverless-stack/open-next/blob/274d446ed7e940cfbe7ce05a21108f4c854ee37a/README.md?plain=1#L66
-      code,
+      ...nodejsFnProps,
+      bundling: {
+        ...nodejsFnProps.bundling,
+        logLevel: LogLevel.SILENT, // silence error on use of `eval` in node_module
+        commandHooks: {
+          afterBundling: () => [],
+          beforeBundling: (_inputDir, outputDir) => [
+            // copy non-bundled assets into zip. use node -e so cross-os compatible
+            `node -e "fs.cpSync('${fixPath(props.nextBuild.nextImageFnDir)}', '${fixPath(
+              outputDir
+            )}', { recursive: true, filter: (src) => !src.includes('/node_modules') && !src.endsWith('index.mjs') })"`,
+          ],
+          beforeInstall: () => [],
+        },
+      },
+      entry: join(props.nextBuild.nextImageFnDir, NEXTJS_BUILD_INDEX_FILE),
       handler: 'index.handler',
-      runtime: LAMBDA_RUNTIME,
-      architecture: Architecture.ARM_64,
       description: 'Next.js Image Optimization Function',
-      // prevents "Resolution error: Cannot use resource in a cross-environment
-      // fashion, the resource's physical name must be explicit set or use
-      // PhysicalName.GENERATE_IF_NEEDED."
-      functionName: Stack.of(scope).region !== 'us-east-1' ? PhysicalName.GENERATE_IF_NEEDED : undefined,
       ...lambdaOptions,
-      // defaults
-      memorySize: lambdaOptions?.memorySize || DEFAULT_LAMBA_MEMORY,
-      timeout: lambdaOptions?.timeout ?? Duration.seconds(10),
       environment: {
         BUCKET_NAME: bucket.bucketName,
+        ...lambdaOptions?.environment,
       },
     });
 
-    this.bucket = bucket;
-    this.addPolicy();
-  }
-
-  /**
-   * Adds policy statement to give GetObject permission Image Optimization lambda.
-   */
-  private addPolicy(): void {
-    const policyStatement = new PolicyStatement({
-      actions: ['s3:GetObject'],
-      resources: [this.bucket.arnForObjects('*')],
-    });
-
-    this.role?.attachInlinePolicy(
-      new Policy(this, 'get-image-policy', {
-        statements: [policyStatement],
-      })
-    );
+    bucket.grantRead(this);
```

Comment:
This is nice, I was confused previously why we had more complex policy definitions. There may be good reasons for them of course, but if we have no specific reason the `bucket.grandRead(this)` syntax is really easy to read and (I think?) a big reason people like CDK.